src/main/scala/com/softwaremill/mqperf/mq/ArtemisMq.scala

```diff
@@ -0,0 +1,18 @@
+package com.softwaremill.mqperf.mq
+
+import javax.jms.ConnectionFactory
+
+import com.softwaremill.mqperf.config.TestConfig
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory
+
+class ArtemisMq(val testConfig: TestConfig) extends Jms2Mq {
+
+  override lazy val connectionFactory: ConnectionFactory = {
+    val hosts = "failover:(" + testConfig.brokerHosts.map(h => s"tcp://$h:61616").mkString(",") + ")"
+    val cf = new ActiveMQConnectionFactory(hosts)
+    cf.setConfirmationWindowSize(10485760)
+    cf.setBlockOnDurableSend(false)
```

Comment:
This is really is just first correcting the basic things.

Right journal type for your use case you have
Ensuring you have the correct client
And to make use of JMS 2.0

There are actually even more tuning you can do, and likewise you are running single master when you can actually run with a cluster multi master (similar to kafka) that allows to load balance (see blog response) to go even further.

So please do let us know the results or update the blog results and if you need more we can probably look at further tunings for you and also setting up multi master cluster with co-located replication ha.